demo-app/src/components/Node.tsx

```diff
@@ -104,24 +113,33 @@ export class Node extends React.Component<NodeProps, {}> {
     /**
      * Creates the Node[] components from given nodes.
      *
-     * @param {NodeProps[]} nodes The nodes to render.
+     * @param {string[]} nodeIds The nodes to render.
      * @param {ParentData} parentData The parent data to pass.
      * @returns {JSX.Element[]} The array of JSX elements with nodes.
      */
-    public static renderSublist(nodes: NodeProps[], parentData: ParentData): JSX.Element[] {
-        if (nodes) {
-            let elements: JSX.Element[] = [];
-            for (let i = 0; i < nodes.length; i++) {
+    public static renderSublist(nodeIds: string[], parentData: ParentData): JSX.Element[] {
+        let elements: JSX.Element[] = [];
+        for (let i = 0; i < nodeIds.length; i++) {
```

Comment:
It might be better no to use `for` here and use `map` wih direct return. You will not have o assign to variables (which is rather expensive) and no need o use push as well. If you are going to use this approach change this code so it will suite you the best in readability style (I really like JS shortcutting, but some find it hard to read).
```TSX
public static renderSublist(nodeIds: string[], parentData: ParentData): JSX.Element[] {
  return nodeIds.map(id => (
    parentData.isRedux && (
      <ConnectedNode
        key={id}
        parentData={parentData}
        nodeId={id}
      />
    )
  ) || (
  <Node
    key={id}
    parentData={parentData}
    nodeId={id}
    {...parentData.tree[id]}
  />
))
}
```

Reply:
I will check it, but it was done this way before (without the if statement) because assigning variables and doing a for-cycle was 8-10 times faster than the map. In case we render 100 thousands of nodes the impact can be huge.